templates/editConn.js

```diff
@@ -1,98 +1,118 @@
-const vscode = acquireVsCodeApi();
-let err = [];
-save();
-testConn();
-
-function save() {
-  document.getElementById("save").addEventListener("click", () => {
-    let data = getData();
-    if (err.length > 0) {
-      return {};
-    }
-    vscode.postMessage({
-      command: "save",
-      connID: data.connID,
-      connName: data.connName,
-      connHost: data.connHost,
-      connToken: data.connToken,
-      connOrg: data.connOrg,
-      connDefault: data.connDefault
-    });
-  });
-}
+class Actions {
+  constructor(hasErr, vscode) {
+    this.hasErr = hasErr;
+    this.vscode = vscode;
+  }
 
-function testConn() {
-  document.getElementById("testConn").addEventListener("click", () => {
-    let data = getData();
-    if (err.length > 0) {
-      return {};
-    }
-    vscode.postMessage({
-      command: "testConn",
-      connName: data.connName,
-      connHost: data.connHost,
-      connToken: data.connToken,
-      connOrg: data.connOrg
+  save() {
+    let self = this;
+    document.getElementById("save").addEventListener("click", () => {
+      let data = self.getData();
+      if (self.hasErr) {
+        return {};
+      }
+      self.vscode.postMessage({
```

Comment:
this is a bound function, you can just use this here and remove the self variable.